src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AbstractEnchantmentMachine.java

```diff
@@ -55,4 +61,23 @@ protected void showEnchantmentLevelWarning(@Nonnull BlockMenu menu) {
         menu.replaceExistingItem(22, progressBar);
     }
 
+    protected boolean hasIgnoredLore(@Nonnull ItemStack itemStack) {
+        List<String> ignoredLore = ignoredLores.getValue();
+        // Skip the check if not set any ignored lore.
+        if (ignoredLore.isEmpty()) {
+            return false;
+        }
+        if (itemStack.hasItemMeta()) {
+            List<String> itemLore = itemStack.getItemMeta().getLore();
+            if (itemLore == null) {
+                return false;
+            }
```

Comment:
You should do `ItemMeta#hasLore()` instead of relying on a null check.

Reply:
Yes.